File: java/Tarea12.java
/*
Nombre: Jesus Martin Guajardo Olivo
Matricula: 1838264
Hora: N2
Descripcion: Generar numeros nones
*/
import javax.swing.JOptionPane;
public class Tarea12{
	public static void main(String[] args) {
		int i,n,l,veces=3;
		String entrada,cartel="Los numeros quedaron: ";
		entrada = JOptionPane.showInputDialog("Ingrese los numeros a mostrar");
		n=Integer.parseInt(entrada);
		for(i=1;i<n+1;i++)
		{
			if(i%2==0)
			{
				for(l=1;l<veces;l++)
				{
					cartel = cartel+i+" ";
					
				}
				veces++;
			}
			else
			{
				cartel=cartel+(i*-1)+" ";
				
			}
		}
		JOptionPane.showMessageDialog(null,cartel);
	}//Fin del metodo main
}//Fin de la clase

File: Ruby/Tarea12.rb
print("Ingrese la cantidad de numeros a mostrar")
n = gets().to_i
cartel = ""
veces = 2
for i in (1...n)
	if (i%2==0)
		for i in (1...veces)
			cartel = cartel + i.to_s+" "
		end
		veces++
	if (i%2!=0)
		j = i*-1
		cartel = cartel + j.to_s+" "
	end
end
end
print("#{carte}")
